manager-rs/directory-service/src/database/backups.rs

```diff
@@ -0,0 +1,161 @@
+use sqlx::sqlite::{SqlitePool, SqliteRow};
+use sqlx::Error as SqlxError;
+use sqlx::{Sqlite, Transaction};
+
+use crate::data::error::DirectoryError;
+
+/// Creates the backup networks table if it does not exist already.
+/// Contains all networks that are used as a backup for this network
+pub async fn init_table(pool: &SqlitePool) -> Result<(), DirectoryError> {
+    sqlx::query(
+        "CREATE TABLE IF NOT EXISTS backups_directory_table (
+            user_id TEXT NOT NULL,
+            backup_network_id TEXT NOT NULL,
+            PRIMARY KEY (user_id, backup_network_id)
+        );",
+    )
+    .execute(pool)
+    .await?;
+    Ok(())
+}
+
+/* Queries */
+
+/// Adds a user with a backup network.
+pub async fn add(
+    transaction: &mut Transaction<'_, Sqlite>,
+    user_id: &str,
+    backup_network_id: &str,
+) -> Result<(), SqlxError> {
+    sqlx::query(
+        "INSERT INTO backups_directory_table
+        VALUES ($1,$2)",
+    )
+    .bind(user_id)
+    .bind(backup_network_id)
+    .execute(transaction)
+    .await?;
+
+    Ok(())
+}
+
+/// Gets all of the backup networks for a user.
+pub async fn get(
+    transaction: &mut Transaction<'_, Sqlite>,
+    user_id: &str,
+) -> Result<Vec<SqliteRow>, SqlxError> {
+    Ok(sqlx::query(
+        "SELECT * FROM backups_directory_table
+        WHERE user_id=$1;",
+    )
+    .bind(user_id)
+    .fetch_all(transaction)
+    .await?)
+}
+
+/// Removes all backup networks for a user.
+pub async fn remove(
+    transaction: &mut Transaction<'_, Sqlite>,
+    user_id: &str,
+) -> Result<(), SqlxError> {
+    sqlx::query(
+        "DELETE FROM backups_directory_table
+        WHERE user_info_id=$1",
```

Comment:
seems like the wrong column name here?

Reply:
Good catch! Forgot to add a test too, but its fixed